src/bindings/python/flux/core/handle.py

```diff
@@ -178,3 +179,42 @@ def get_rank(self):
 
     def attr_get(self, attr_name):
         return self.flux_attr_get(attr_name).decode("utf-8")
+
+    def reactor_run(self, reactor=None, flags=0):
+        """
+        Run reactor associated with this Flux handle or reactor argument
+        if it is provided. Sets a signal watcher for SIGINT to return
+        from the reactor on Ctrl-C, and raise KeyboardInterrupt.
+        """
+        if reactor is None:
+            reactor = self.get_reactor()
+
+        reactor_interrupted = False
+
+        def reactor_interrupt(h, *args):
+            #  ensure reactor_interrupted from enclosing scope:
+            nonlocal reactor_interrupted
+            reactor_interrupted = True
+            h.reactor_stop(reactor)
+
+        with self.signal_watcher_create(signal.SIGINT, reactor_interrupt):
+            rc = self.flux_reactor_run(self.get_reactor(), flags)
```

Comment:
shouldn't this be `rc = self.flux_reactor_run(reactor, flags)`, you set `reactor = self.get_reactor()` above.

Reply:
Yes, good catch! That is bug because `flux_reactor_run()` could be called on the wrong reactor if one was passed in. (Not that we do that in python anywhere)